src/client.rs

```diff
@@ -877,6 +877,19 @@ where
 
                         self.buffer.put(&original[..]);
 
+                        let mut final_buffer = self.buffer.clone();
+                        let first_message_code = final_buffer.get_u8() as char;
+                        let _ = final_buffer.get_i32() as usize;
+
+                        // Almost certainly true
+                        if first_message_code == 'P' {
```

Comment:
Why not run this check in `'P' => {` block? Should be cheaper than cloning the entire buffer (which can be very large) every `S` (effectively every query).

Reply:
Also, aren't all Postgres strings NULL-terminated, so this check would fire for named prepared statements as well?

Reply:
Good point on the clone, the reason I run this in the `S` and not in the `P` is that we have two places where we handle `P` buffering so I am trying to avoid duplication. Let me see if we can get away with no clones.

re: `NULL-terminated`, we only care about the first byte after `P` and the 32bit size. If it is `\0` the this is an unnamed prepared statement, anything else would be interpreted as a named prepared statement.